tensorpipe/test/channel/channel_test.cc

```diff
@@ -17,23 +17,21 @@ using namespace tensorpipe::channel;
 // because we need this test case to run in a subprocess as in some cases it may
 // initialize CUDA and thus would otherwise "pollute" the parent process.
 template <typename TBuffer>
-class DomainDescriptorTest : public ClientServerChannelTestCase<TBuffer> {
+class DomainDescriptorTest : public ChannelTestCase<TBuffer> {
  public:
-  void server(std::shared_ptr<transport::Connection> /* unused */) override {
-    this->peers_->done(PeerGroup::kServer);
-    this->peers_->join(PeerGroup::kServer);
-  }
-
-  void client(std::shared_ptr<transport::Connection> /* unused */) override {
-    std::shared_ptr<Context<TBuffer>> context1 =
-        this->helper_->makeContext("ctx1");
-    std::shared_ptr<Context<TBuffer>> context2 =
-        this->helper_->makeContext("ctx2");
-    EXPECT_FALSE(context1->domainDescriptor().empty());
-    EXPECT_FALSE(context2->domainDescriptor().empty());
-    EXPECT_EQ(context1->domainDescriptor(), context2->domainDescriptor());
-    this->peers_->done(PeerGroup::kClient);
-    this->peers_->join(PeerGroup::kClient);
+  void run(ChannelTestHelper<TBuffer>* helper) override {
+    auto peerGroup = helper->makePeerGroup();
+    peerGroup->spawn(
+        [&] {
+          std::shared_ptr<Context<TBuffer>> context1 =
+              helper->makeContext("ctx1");
+          std::shared_ptr<Context<TBuffer>> context2 =
+              helper->makeContext("ctx2");
+          EXPECT_FALSE(context1->domainDescriptor().empty());
+          EXPECT_FALSE(context2->domainDescriptor().empty());
+          EXPECT_EQ(context1->domainDescriptor(), context2->domainDescriptor());
```

Comment:
Just noticed (and unrelated to this diff): do we want to now change this to check that `context1->canCommunicateWithRemote(context2->domainDescriptor())`?

Reply:
Good question. In its current form, this test somewhat ensures that the domain descriptor is the same in subsequent calls. I'm not sure which property we want to test.

Reply:
Well we can't really check for any "deep" semantic meaning of the descriptor (like, that it compares true or false in the various scenarios), as that varies for each channel and it requires specific test setups. So I guess we're mainly checking for trivial properties, and "reflexivity" seems a good one. However, I also now think that ensuring that all contexts started in the same conditions should produce the same descriptor. So maybe we should both check for equality, and that `canCommunicateWithRemote` is true?